modules/data-table/main/index.coffee

```diff
@@ -15,6 +21,203 @@ hx.userFacingText({
 fullWidthColSpan = 999 # the colspan used to make a cell display as an entire row
 collapseBreakPoint = 480
 
+columnOptionLookup = (options, name, id) ->
+  if options.columns isnt undefined and options.columns[id] isnt undefined and options.columns[id][name] isnt undefined
+    options.columns[id][name]
+  else
+    options[name]
+
+splitArray = (array, index) ->
+  left = if index is 0
+    []
+  else
+    array[0..index - 1]
```

Comment:
Actually, maybe this would be clearer without the newlines?

```
left = if index is 0 then [] else array[0...index]
right = if index is array.length - 1 then [] else array[index+1...array.length]
```